src/pybind/mgr/cephadm/serve.py

```diff
@@ -731,6 +731,11 @@ def _check_daemons(self) -> None:
                     action = 'redeploy'
                 try:
                     daemon_spec = CephadmDaemonDeploySpec.from_daemon_description(dd)
+                    if (action == 'reconfig' and self.mgr.get_active_mgr_digests()
+                            and (not dd.deployed_by or not any(d in dd.deployed_by for d in (self.mgr.get_active_mgr_digests() or [])))):
+                        self.log.info(
+                            "Delaying reconfiguration of %s daemon until it has been deployed by mgr with correct version", dd.name())
```

Comment:
Imagine we move all MONs to a completely different hosts while the upgrade is still in progress and then restart the MGR. Can we avoid delaying the reconfig, till the MGR can no longer join the cluster?